dask_jobqueue/tests/test_remote_slurm.py

```diff
@@ -0,0 +1,57 @@
+import sys
+
+import pytest
+from distributed import Client
+
+from dask_jobqueue.remote_slurm import RemoteSLURMCluster
+
+TEST_API_URL = "http://a-url.com/slurm"
+
+
+@pytest.mark.env("slurm")
+@pytest.mark.asyncio
+async def test_end_to_end(aresponses, loop):
+    aresponses.add(TEST_API_URL, "/jobs/submit", "POST", {"job_id": 1001})
+    aresponses.add(
+        TEST_API_URL,
+        "/job/1001",
+        "DELETE",
+        aresponses.Response(status=200),
+    )
+
+    async with RemoteSLURMCluster(
+        api_url=TEST_API_URL,
+        walltime="00:02:00",
+        processes=4,
+        cores=8,
+        memory="28GB",
+        n_workers=1,
+        job_extra=["--gpus 2"],
+        loop=loop,
+        asynchronous=True
+    ) as cluster:
+        await cluster
+        assert aresponses.history[0].request.json() == {
+            "script": "#!/usr/bin/env bash\n"
+            "\n"
+            "#SBATCH -J dask-worker\n"
+            "#SBATCH -n 1\n"
+            "#SBATCH --cpus-per-task=8\n"
+            "#SBATCH --mem=27G\n"
+            "#SBATCH -t 00:02:00\n"
+            "#SBATCH --gpus 2\n"
+            "\n"
+            f"{sys.executable} -m "
+            f"distributed.cli.dask_worker {cluster.scheduler.address} --nthreads "
+            "2 --nprocs 4 --memory-limit 7.00GB --name RemoteSLURMCluster-0 "
+            "--nanny --death-timeout 60 --protocol tcp://\n",
+        }
```

Comment:
I assume this could subtly change over time and the test would need to be updated each time.

It may be better to replace this with a series of assertions testing some of the content.

```suggestion
        request = aresponses.history[0].request.json()
        assert  "#SBATCH -J dask-worker" in request
        assert  "#SBATCH --gpus 2" in request
        assert  sys.executablein request
        assert  "--name RemoteSLURMCluster-0" in request
        # etc, etc
```

Reply:
Good point. I'll probably completely remove this test as the `aresponses` dependency cannot be installed by conda and therefore it will never execute on CI.